components/pedigree/resources/src/main/resources/pedigree/view/workspace.js

```diff
@@ -139,17 +140,17 @@ define([
             // set display:block
             svgText = svgText.replace(/(<svg[^<>]+style=")/g, "$1display:block; ");
```

Comment:
DOM manipulation of a rendered DOM is expensive, since each change causes
the page to re-render. An in-memory DOM subtree that has not been added to
the page is a bit less expensive. What you could also do is use a "virtual
DOM" library like mithril.js to generate a virtual DOM, which then you
render to actual dom elements when nescessary. (http://mithril.js.org/)

I'm just talking about all the hypothetical ways to "skin a cat." You did
say that in this particular usecase, the performance isn't a huge issue
since it's only done once, so I doubt we need to involve a whole new
library in this :P

On Thu, Nov 19, 2015 at 10:14 PM, allasm <notifications@github.com> wrote:

> In
> components/pedigree/resources/src/main/resources/pedigree/view/workspace.js
> <https://github.com/phenotips/phenotips/pull/1927#discussion_r45432199>:
>
> > @@ -139,17 +140,17 @@ define([
> >              // set display:block
> >              svgText = svgText.replace(/(<svg[^<>]+style=")/g, "$1display:block; ");
>
> I'm afraid with a large SVG DOM may be visibly slow - but that needs to be
> tested (in general from my experience DOM manipulation is MUCH slower than
> text manipulation, while being safer/less prone to mistakes, of course -
> the slow part is not to clone, but to do actual modifications) - especially
> as for say print preview we need effectively N copies of SVG, with
> different viewports (then again, can do an in-memory copy, do DOM
> manipulations, and only change vieports as text)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phenotips/phenotips/pull/1927/files#r45432199>.
>



-- 
Denys Pavlov